tools/monitoring/agents/coreosprometheusoperator/lib.go

```diff
@@ -28,17 +29,14 @@ func New(k8sClient kubernetes.Interface, extClient ecs.ApiextensionsV1beta1Inter
 }
 
 func (agent *PrometheusCoreosOperator) Add(sp api.StatsAccessor, spec *api.AgentSpec) error {
-	if !agent.SupportsCoreOSOperator() {
-		return errors.New("cluster does not support CoreOS Prometheus operator")
-	}
-	return agent.ensureServiceMonitor(sp, spec, spec)
+	return agent.Update(sp, spec)
 }
 
-func (agent *PrometheusCoreosOperator) Update(sp api.StatsAccessor, old, new *api.AgentSpec) error {
+func (agent *PrometheusCoreosOperator) Update(sp api.StatsAccessor, new *api.AgentSpec) error {
 	if !agent.SupportsCoreOSOperator() {
 		return errors.New("cluster does not support CoreOS Prometheus operator")
 	}
-	return agent.ensureServiceMonitor(sp, old, new)
+	return agent.ensureServiceMonitor(sp, new)
 }
 
 func (agent *PrometheusCoreosOperator) Delete(sp api.StatsAccessor, spec *api.AgentSpec) error {
```

Comment:
Also remove `spec *api.AgentSpec)` from Delete, and use the label to find out the ServiceMonitor and delete it